File: src/kowalski/cmd/add.py
import sys
import os

from datetime    import datetime
from frontmatter import load

from kowalski.internal.config import load_config
from kowalski.internal.run    import execute
from kowalski.internal.note   import Note


def add_cmd(content: str, tags: list[str]):

    PATH, EDITOR = load_config()

    if not sys.stdin.isatty():
        piped_content = sys.stdin.read().strip()
        content = piped_content if piped_content else content

    timestamp = datetime.now().replace(microsecond=0).isoformat()

    name = f"{timestamp}.md"
    created = timestamp
    tags = tags

    if content is None:

        note = Note(name, created, tags, "")
        note.write(PATH)

        note_path = os.path.join(PATH, name)
        execute(f"{EDITOR} {note_path}")

        md = load(note_path)
        tags = md["tags"]
        content = md.content

    note = Note(name, created, tags, content)
    note.write(PATH)

    print(note.format())

File: pkg/add/add.go
package add

import (
	"fmt"
	"github.com/Zatfer17/kowalski-core/internal/config"
	"github.com/Zatfer17/kowalski-core/internal/model"
)

func Add(created string, tags []string, content string) (model.Note, error) {

	config, err := config.InitConfig()
	if err != nil {
		return model.Note{}, fmt.Errorf("error reading config file: %w", err)
	}

	n := model.Note{Created: created, Tags: tags, Content: content}
	
	if err := n.Write(config.NotesPath); err != nil {
		return model.Note{}, fmt.Errorf("error saving note: %w", err)
	}

	return n, nil
}